client/src/utils/getCompressParams.ts

```diff
@@ -0,0 +1,56 @@
+import { TPortalSystem, TTapisFile } from './types';
+
+export const getCompressParams = (
+  files: TTapisFile[],
+  archiveFileName: string,
+  compressionType: string,
+  compressApp: { id: string; version: string },
+  defaultAllocation: string,
+  defaultPrivateSystem?: TPortalSystem
+) => {
+  const fileInputs = files.map((file) => ({
+    sourceUrl: `tapis://${file.system}/${file.path}`,
+  }));
+
+  let archivePath, archiveSystem;
+
+  if (defaultPrivateSystem) {
+    archivePath = defaultPrivateSystem.homeDir;
+    archiveSystem = defaultPrivateSystem.system;
+  } else {
+    archivePath = `${files[0].path.slice(0, -files[0].name.length)}`;
+    archiveSystem = files[0].system;
+  }
```

Comment:
Looking at this again, I think we want to always archive to the file.path and file.system of the file being compressed, rather than the defaultPrivateSystem root

Reply:
I'll ditch the `if/else` statement and only execute the `else` portion above.